apps/arweave/src/ar_packing_server.erl

```diff
@@ -243,13 +245,30 @@ encipher_replica_2_9_chunk(Chunk, Entropy) ->
 ) -> binary().
 generate_replica_2_9_entropy(RewardAddr, BucketEndOffset, SubChunkStartOffset) ->
 	Key = ar_replica_2_9:get_entropy_key(RewardAddr, BucketEndOffset, SubChunkStartOffset),
-	PackingState = get_packing_state(),
-	RandomXState = get_randomx_state_by_packing({replica_2_9, RewardAddr}, PackingState),
 
-	Entropy = ar_mine_randomx:randomx_generate_replica_2_9_entropy(RandomXState, Key),
-	%% Primarily needed for testing where the entropy generated exceeds the entropy
-	%% needed for tests.
-	binary_part(Entropy, 0, ?REPLICA_2_9_ENTROPY_SIZE).
+	entropy_generation_lock(Key, RewardAddr, BucketEndOffset, SubChunkStartOffset),
+	case ar_replica_2_9_entropy_cache:get(Key) of
```

Comment:
Is the entropy cache needed now to handle 2 chunks that need to be packed to the same entropy but one chunk comes from `normal` syncing and one from `footprint` syncing? Or asked another way: why do we need the entropy cache? when do we expect there to be cache hit?

Reply:
I see some possibilities when the cache may be not hit:

- we cannot prepare the entropy fast enough for the chunks fetched during the "normal" sync (they are fetched unpacked);
- we process several footprints at a time (perhaps, collecting small leftovers; perhaps, syncing too fast)